### ISSUE TITLE
Encountered an exception by network error.

### ISSUE THREAD

**Issue by otolab** [GoogleCloudPlatform/gcloud-node#918]
My utility has encountered an exception. It's occurred in the "libs/common/utils.js" by network error.

Looks like a regression from https://github.com/GoogleCloudPlatform/gcloud-node/commit/8b2bbf2d5f9da7cceab4d544cf6983e27d2ad534#diff-638107c62532c8d249c9f17a0c7fb355L169

`resp` check has been removed. therefore, It throws an exception when `httpRespMessage` is `undefined`.

I wrote fix patch. please see pull request.

**Comment by dg3feiko** [comment]
I encounter the same problem when I did insert to Bigquery tables

```
TypeError: Cannot read property 'statusCode' of undefined
    at Object.parseHttpRespMessage (/home/deploy/streamer/node_modules/gcloud/lib/common/util.js:168:22)
    at Object.handleResp (/home/deploy/streamer/node_modules/gcloud/lib/common/util.js:145:10)
    at /home/deploy/streamer/node_modules/gcloud/lib/common/util.js:447:12
    at Request.onResponse [as _callback] (/home/deploy/streamer/node_modules/gcloud/node_modules/retry-request/index.js:91:9)
    at self.callback (/home/deploy/streamer/node_modules/gcloud/node_modules/request/request.js:198:22)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at Request.onRequestError (/home/deploy/streamer/node_modules/gcloud/node_modules/request/request.js:867:8)
    at emitOne (events.js:77:13)
    at ClientRequest.emit (events.js:169:7)
    at TLSSocket.socketErrorListener (_http_client.js:259:9)
    at emitOne (events.js:77:13)
    at TLSSocket.emit (events.js:169:7)
    at onwriteError (_stream_writable.js:304:10)
    at onwrite (_stream_writable.js:322:5)
    at WritableState.onwrite (_stream_writable.js:89:5)
```

**Comment by stephenplusplus** [comment]
Thanks for reporting! A patch will be out soon.

**Comment by stephenplusplus** [comment]
Released as v0.24.1. Thanks again!
